Index: weston-1.5.0/desktop-shell/shell.c
===================================================================
--- weston-1.5.0.orig/desktop-shell/shell.c	2014-05-20 02:00:51.000000000 +0300
+++ weston-1.5.0/desktop-shell/shell.c	2014-10-01 07:11:38.480248533 +0300
@@ -3114,8 +3114,6 @@
 static void
 destroy_shell_surface(struct shell_surface *shsurf)
 {
-	struct shell_surface *child, *next;
-
 	wl_signal_emit(&shsurf->destroy_signal, shsurf);
 
 	if (!wl_list_empty(&shsurf->popup.grab_link)) {
@@ -3139,8 +3137,6 @@
 	weston_view_destroy(shsurf->view);
 
 	wl_list_remove(&shsurf->children_link);
-	wl_list_for_each_safe(child, next, &shsurf->children_list, children_link)
-		shell_surface_set_parent(child, NULL);
 
 	wl_list_remove(&shsurf->link);
 	free(shsurf);
